cachelib/cachebench/util/Request.h

```diff
@@ -60,6 +60,35 @@ enum class OpResultType {
   kCouldExistFalse
 };
 
+struct BinaryRequest {
+  uint8_t keySize_;
+  uint8_t op_;
+  uint16_t repeats_;
+  uint16_t ttl_;
+  uint64_t keyOffset_;
+  uint32_t valueSize_;
+
+  BinaryRequest() = default;
+
+  BinaryRequest(uint8_t keySize,
+                size_t valueSize,
+                uint16_t repeats,
+                uint8_t op,
+                uint16_t ttl,
+                size_t keyOffset)
+      : keySize_(keySize),
+        valueSize_(valueSize),
+        repeats_(repeats),
+        op_(op),
+        ttl_(ttl),
+        keyOffset_(keyOffset) {}
+
+  std::string_view getKey(char* baseAddr) const {
```

Comment:
if you will store offset relative to `this` pointer than you do not need to pass `baseAddr` as a parameter. The code of the method will look something like this:
```
std::string_view getKey() const {
    return std::string_view(
        reinterpret_cast<const char*>(this+ keyOffset_), keySize_);
}
```

Reply:
I just figured out that you store keys in separate file, correct? if so, my idea is not feasible. It is only possible if `BinaryRequest` objects and corresponding keys are stored in the same file.

Reply:
done - thanks for the note!